### PR TITLE
Resolve Xcode 9.3 warnings and deprecations

### LINKED ISSUE #219: Warnings on XCode 9.3
Hi, team. I'm a first-time FLEX user and love it so far. I'm using XCode 9.3 and am experiencing the following warnings at build time. They don't seem to cause any negative impact.

`.../Pods/FLEX/Classes/ExplorerInterface/FLEXExplorerViewController.m:168:1: Implementing deprecated method
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.3.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIViewController.h:327:1: Method 'willRotateToInterfaceOrientation:duration:' declared here
.../Pods/FLEX/Classes/ExplorerInterface/FLEXExplorerViewController.m:176:1: Implementing deprecated method
....../iPhoneSimulator11.3.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIViewController.h:328:1: Method 'didRotateFromInterfaceOrientation:' declared here
.../Pods/FLEX/Classes/ExplorerInterface/FLEXExplorerViewController.m:822:29: This block declaration is not a prototype
.../Pods/FLEX/Classes/ExplorerInterface/FLEXExplorerViewController.m:846:29: This block declaration is not a prototype
.../Pods/FLEX/Classes/Utility/FLEXKeyboardShortcutManager.m:169:83: 'maximumPossibleForce' is only available on iOS 9.0 or newer
....../iPhoneSimulator11.3.sdk/System/Library/Frameworks/UIKit.framework/Headers/UITouch.h:70:39: 'maximumPossibleForce' has been explicitly marked partial here
.../Pods/FLEX/Classes/Utility/FLEXKeyboardShortcutManager.m:169:83: Enclose 'maximumPossibleForce' in an @available check to silence this warning
.../Pods/FLEX/Classes/Network/PonyDebugger/FLEXNetworkObserver.m:677:53: This block declaration is not a prototype
.../Pods/FLEX/Classes/Network/PonyDebugger/FLEXNetworkObserver.m:758:51: This block declaration is not a prototype
.../Pods/FLEX/Classes/GlobalStateExplorers/DatabaseBrowser/FLEXTableContentViewController.m:174:13: Block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior
.../Pods/FLEX/Classes/GlobalStateExplorers/DatabaseBrowser/FLEXTableContentViewController.m:177:13: Block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior`

### PR DESCRIPTION
**Pull request by bharat**
This pull request should resolve #219

### CODE REVIEW

**Comment by bharat** [comment]
not sure why my last commit is failing to build. I'm assuming it has something to do with the version of Xcode that we're testing against with Travis? open to ideas here.

**Comment by matrush** [comment]
Please use `FLEX_AT_LEAST_IOS11_SDK` to wrap the `@available` since the syntax is added in Xcode 9.0. The CI still builds on Xcode 8 so it will fail.

File: Classes/ExplorerInterface/FLEXExplorerViewController.m [hunk 1]
```diff
@@ -695,15 +695,15 @@ - (void)updateSelectedViewPositionWithDragGesture:(UIPanGestureRecognizer *)move
 - (CGRect)viewSafeArea
 {
     CGRect safeArea = self.view.bounds;
-#if FLEX_AT_LEAST_IOS11_SDK
+#ifdef FLEX_AT_LEAST_IOS11_SDK
```
Comment (matrush) [inline Classes/ExplorerInterface/FLEXExplorerViewController.m]: keep `#if`

File: Classes/Toolbar/FLEXExplorerToolbar.m [hunk 2]
```diff
@@ -261,7 +261,7 @@ - (CGSize)sizeThatFits:(CGSize)size
 - (CGRect)safeArea
 {
   CGRect safeArea = self.bounds;
-#if FLEX_AT_LEAST_IOS11_SDK
+#ifdef FLEX_AT_LEAST_IOS11_SDK
```
Comment (matrush) [inline Classes/Toolbar/FLEXExplorerToolbar.m]: keep `#if`

File: Classes/ExplorerInterface/FLEXExplorerViewController.m [hunk 3]
```diff
@@ -695,15 +695,15 @@ - (void)updateSelectedViewPositionWithDragGesture:(UIPanGestureRecognizer *)move
...
     if (@available(iOS 11, *)) {
         safeArea = UIEdgeInsetsInsetRect(self.view.bounds, self.view.safeAreaInsets);
     }
 #endif
     return safeArea;
 }
 
-#if FLEX_AT_LEAST_IOS11_SDK
+#ifdef FLEX_AT_LEAST_IOS11_SDK
```
Comment (matrush) [inline Classes/ExplorerInterface/FLEXExplorerViewController.m]: Keep `#if`

File: Classes/ExplorerInterface/FLEXExplorerViewController.m [hunk 4]
```diff
@@ -171,32 +171,32 @@ - (BOOL)shouldAutorotate
...
-    }
-    
-    if (self.selectedView) {
-        self.selectedViewOverlay.frame = [self frameInLocalCoordinatesForView:self.selectedView];
-        self.selectedViewOverlay.hidden = NO;
-    }
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
+{
+    [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context)
+     {
```
Comment (matrush) [inline Classes/ExplorerInterface/FLEXExplorerViewController.m]: let's keep this `{` in L176 per code style convention

File: Classes/ExplorerInterface/FLEXExplorerViewController.m [hunk 5]
```diff
@@ -171,32 +171,32 @@ - (BOOL)shouldAutorotate
...
+         for (UIView *outlineView in [self.outlineViewsForVisibleViews allValues]) {
+             outlineView.hidden = YES;
+         }
+         self.selectedViewOverlay.hidden = YES;
+     } completion:^(id<UIViewControllerTransitionCoordinatorContext> context)
+     {
```
Comment (matrush) [inline Classes/ExplorerInterface/FLEXExplorerViewController.m]: same

File: Classes/Utility/FLEXKeyboardShortcutManager.m [hunk 6]
```diff
@@ -165,9 +165,11 @@ + (void)load
                     pressureLevel++;
                 }
                 if (pressureLevel > 0) {
-                    for (UITouch *touch in [event allTouches]) {
-                        double adjustedPressureLevel = pressureLevel * 20 * touch.maximumPossibleForce;
-                        [touch setValue:@(adjustedPressureLevel) forKey:@"_pressure"];
+                    if (@available(iOS 9.0, *)) {
```
Comment (matrush) [inline Classes/Utility/FLEXKeyboardShortcutManager.m]: you need to add the macro check `FLEX_AT_LEAST_IOS11_SDK` here as well to make it build with CI Xcode 8

File: Classes/Utility/FLEXUtility.h [hunk 7]
```diff
@@ -14,7 +14,9 @@
 
 #define FLEXFloor(x) (floor([[UIScreen mainScreen] scale] * (x)) / [[UIScreen mainScreen] scale])
 
-#define FLEX_AT_LEAST_IOS11_SDK defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
+#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
```
Comment (matrush) [inline Classes/Utility/FLEXUtility.h]: Let's change it to like this so you don't need to modify other files:
```
#if defined(__IPHONE_11_0)
#define FLEX_AT_LEAST_IOS11_SDK (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
#else
#define FLEX_AT_LEAST_IOS11_SDK NO
#endif
```

File: Example/UICatalog.xcodeproj/xcshareddata/xcschemes/UICatalog.xcscheme [hunk 8]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0800"
+   LastUpgradeVersion = "0930"
```
Comment (matrush) [inline Example/UICatalog.xcodeproj/xcshareddata/xcschemes/UICatalog.xcscheme]: We will want to keep this for Xcode 8 for now. So you can revert the changes to this file.

File: FLEX.xcodeproj/xcshareddata/xcschemes/FLEX.xcscheme [hunk 9]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0900"
+   LastUpgradeVersion = "0930"
```
Comment (matrush) [inline FLEX.xcodeproj/xcshareddata/xcschemes/FLEX.xcscheme]: same

**Comment by bharat** [comment]
Sorry for the long delay. Pushed an update just now! Let me know what other steps you'd like to see.

**Comment by NSExceptional** [comment]
Looks good to me!
